packages/blockchain-extension/cucumber/helpers/walletAndIdentityHelper.ts

```diff
@@ -123,19 +124,29 @@ export class WalletAndIdentityHelper {
     }
 
     private setIdentityStubs(method: string, identityName: string, mspid: string): void {
-        this.userInputUtilHelper.inputBoxStub.withArgs('Provide a name for the identity').resolves(identityName);
         this.userInputUtilHelper.inputBoxStub.withArgs('Enter MSPID').resolves(mspid);
 
         if (method === 'certs') {
             this.userInputUtilHelper.showAddIdentityMethodStub.resolves(UserInputUtil.ADD_CERT_KEY_OPTION);
+            this.userInputUtilHelper.inputBoxStub.withArgs('Provide a name for the identity').resolves(identityName);
             this.userInputUtilHelper.showGetCertKeyStub.resolves({ certificatePath: WalletAndIdentityHelper.certPath, privateKeyPath: WalletAndIdentityHelper.keyPath });
         } else if (method === 'JSON file') {
             const jsonPath: string = process.env.OPSTOOLS_FABRIC ? path.join(process.env.JSON_DIR, `${identityName}.json`) : WalletAndIdentityHelper.jsonFilePath;
             this.userInputUtilHelper.showAddIdentityMethodStub.resolves(UserInputUtil.ADD_JSON_ID_OPTION);
+            fs.readFile(jsonPath, 'utf8', function readFileCallback(err: any, data: string): void {
+                if (!err) {
+                    const obj: any = JSON.parse(data);
+                    obj.name = identityName;
+                    const json: any = JSON.stringify(obj);
+                    fs.writeFile(jsonPath, json, 'utf8', () => {
+                        return;
+                    });
+            }});
```

Comment:
This worries me, because we shouldn't need to read/modify/write our identities. If the read or the write fail the test will fail for an unrelated reason.

Instead we should make sure that our tests are correct and relevant, which in this case *might* mean renaming the identities we are expecting to create.

I'm happy to discuss this further via DM.